datajunction-server/datajunction_server/internal/nodes.py

```diff
@@ -424,25 +411,34 @@ def save_node(
     session.refresh(node.current)
 
 
+def add_lineage_to_node(session: Session, node_revision: NodeRevision):
```

Comment:
I think a better way is to "never" mutate the arguments of the function. I would write it this way:
```
def get_node_lineage(session: Session, node_revision: NodeRevision):
    ...
    return lineage or []
```
And then use it:
```
some_node_revision.lineage = get_node_lineage(session, some_node_revision)
```
I know we violated this all over the place here, but since you are doing this cleanup already... maybe we can start making small steps?

Reply:
Good point! Also I think this change could just be made on the upstream function `get_column_level_lineage` rather than in a separate function.

Reply:
@agorajek Done! I removed `add_lineage_to_node` and just pushed the conditional checks into the upstream function.